File: src/ELibrary.Frontend/src/app/modules/shared/domain/models/shop/bookstock/stockBookOrder.ts
import { Client, getDefaultClient, getDefaultStockBookChange, StockBookChange } from "../../../..";

export enum StockBookOrderType {
    StockReplenishment, ClientOrder, ClientOrderCancel, ManagerOrderCancel
}
export interface StockBookOrder {
    id: number,
    createdAt: Date,
    updatedAt: Date,
    type: StockBookOrderType,
    totalChangeAmount: number,
    client: Client,
    stockBookChanges: StockBookChange[]
}

export function getDefaultStockBookOrder(): StockBookOrder {
    return {
        id: 0,
        createdAt: new Date(),
        updatedAt: new Date(),
        type: StockBookOrderType.ClientOrder,
        totalChangeAmount: 0,
        client: getDefaultClient(),
        stockBookChanges: [getDefaultStockBookChange()]
    }
}

export function stockBookOrderTypeToString(type: StockBookOrderType): string {
    switch (type) {
        case StockBookOrderType.StockReplenishment:
            return "Stock Replenishment"
        case StockBookOrderType.ClientOrder:
            return "Client Order"
        case StockBookOrderType.ClientOrderCancel:
            return "Client Order Cancel"
        case StockBookOrderType.ManagerOrderCancel:
            return "Manager Order Cancel"
        default:
            return "";
    }
}

File: src/ELibrary.Backend/LibraryShopEntities/Domain/Entities/Shop/StockBookOrder.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LibraryShopEntities.Domain.Entities.Shop
{
    public enum StockBookOrderType
    {
        StockReplenishment, ClientOrder, ClientOrderCancel
    }
    public class StockBookOrder : ITrackable
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        [Required]
        public DateTime CreatedAt { get; set; }
        [Required]
        public DateTime UpdatedAt { get; set; }
        [Required]
        public int TotalChangeAmount { get; set; }
        [Required]
        public StockBookOrderType Type { get; set; }
        [Required]
        public string ClientId { get; set; } = default!;
        [ForeignKey("ClientId")]
        public Client Client { get; set; } = default!;
        public List<StockBookChange> StockBookChanges { get; set; } = new List<StockBookChange>();
    }
}
